src/storage/txn/scheduler.rs

```diff
@@ -1371,39 +1355,37 @@ impl<E: Engine, L: LockManager> TxnScheduler<E, L> {
                 .inc_by(quota_delay.as_micros() as u64);
         }
 
+        write_result
+    }
+
+    /// Process the locks encountered or released by the write command, and try
+    /// to write pessimistic locks in the memory if `in-memory` lock is
+    /// enabled. Return updated `WriteResult` if there are still data to be
+    /// persisted, or None if there is nothing left to do.
+    fn handle_lock_info<'a>(
```

Comment:
It looks not only handling the lock info, but everything about the WriteResult except persisting the data. It looks not very proper to let this function be responsible to determine whether there's anything to be write to the engine. How about naming it `handle_non_persistent_write_result`?